fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/MutableRecordStoreState.java

```diff
@@ -122,13 +122,34 @@ public IndexState setState(@Nonnull String indexName, @Nonnull IndexState state)
         if ((users.get() & WRITE_MASK) == 0) {
             throw new RecordCoreException("record store state is not enabled for modification");
         }
-        IndexState previous;
```

Comment:
Shouldn't this method simply be:

    @Nonnull
    public IndexState setState(@Nonnull String indexName, @Nonnull IndexState state)  {
        IndexMetaData previous = setIndexMetaData(indexName, new IndexMetaData(state));
        return previous == null ? null : previous.getIndexState();
    }